spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/ReactiveRedisStreamMessageHandler.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.redis.outbound;
+
+import org.springframework.data.redis.connection.ReactiveRedisConnectionFactory;
+import org.springframework.data.redis.connection.stream.ObjectRecord;
+import org.springframework.data.redis.connection.stream.StreamRecords;
+import org.springframework.data.redis.core.ReactiveRedisTemplate;
+import org.springframework.data.redis.core.ReactiveStreamOperations;
+import org.springframework.data.redis.hash.HashMapper;
+import org.springframework.data.redis.serializer.RedisSerializationContext;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.handler.AbstractReactiveMessageHandler;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * Implementation of {@link org.springframework.messaging.ReactiveMessageHandler} which writes
+ * Message payload or Message itself(see {@link #extractPayload}) into a Redis stream using Reactive Stream operations.
+ *
+ * @author Attoumane Ahamadi
+ *
+ * @since 5.4
+ */
+@SuppressWarnings({"rawtypes", "unchecked"})
```

Comment:
May we kinda get rid of `rawtypes`?
Probably simple wildcard for genercis (`<?>`) should be enough for us...

Reply:
Removing `rawtypes` causes compilation issue: 
```
error: no suitable method found for opsForStream(HashMapper<CAP#1,CAP#2,CAP#3>)
				template.opsForStream(this.hashMapper);
				        ^
    method ReactiveRedisOperations.<HK#1,HV#1>opsForStream() is not applicable
      (cannot infer type-variable(s) HK#1,HV#1
        (actual and formal argument lists differ in length))
```
on 
```
this.reactiveStreamOperations = this.hashMapper == null ? template.opsForStream() :
				template.opsForStream(this.hashMapper);
```

Reply:
OK. Let me play with that locally!
Then I'll merge it since we need something for today's release.

You feel free to go ahead with Inbound Channel Adapter implementation and some docs for both of this.

Thanks